app/api/v1/coffee_shops_route.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class CoffeeShopsRoute < BaseRoute
+      route do |r|
+        r.get do
+          csv_url = ENV.fetch('COFFEE_SHOPS_CSV_URL', 'https://raw.githubusercontent.com/Agilefreaks/test_oop/master/coffee_shops.csv')
+          user_coordinates = permit_params(r.params, %w[x y]).transform_keys(&:to_sym)
+          csv_data = Services::FetchCsvService.new(csv_url).call
+          coffee_shops = Services::ParseCsvService.new(csv_data).call
+          coffee_shop_service = Services::CoffeeShopService.new(user_coordinates, coffee_shops)
```

Comment:
what's the point of leveraging command pattern here?

Reply:
The command pattern is used to encapsulate the fetching, parsing, and processing of CSV data into separate service objects. Each service object has a single responsibility, it's easier to test and reuse like this. Another pattern that could be use there would be the Strategy pattern but I thought is to complex for this specific problem.

Reply:
![Screenshot_2025-01-28-03-03-20-13_e4424258c8b8649f6e67d283a50a2cbc.jpg](https://github.com/user-attachments/assets/7051a6b0-32bc-4ba3-ae91-86749b1e8ab9)

Ok, but the main need for command pattern is missing here

Reply:
Yes, I think a better pattern would be Service Object Workflow or Facade Pattern.
The client only calls CoffeeShopWorkflowService, reducing its responsibilities, but  still maintain SRP within the individual services FetchCsvService, ParseCsvService, making them testable and reusable.
Future concerns like caching, throttling, or circuit breaking can be added inside the CoffeeShopWorkflowService without affecting the client or individual services.